tcms/rpc/api/forms/testrun.py

```diff
@@ -52,9 +43,35 @@ class UpdateForm(NewForm):
         }
     )
 
+    class Meta:
+        model = TestRun
+        exclude = ('tag', 'cc')  # pylint: disable=modelform-uses-exclude
+
+    def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
+                 initial=None, error_class=ErrorList, label_suffix=None,
+                 empty_permitted=False, instance=None, use_required_attribute=None,
+                 renderer=None):
+        super().__init__(
+            data, files, auto_id, prefix,
+            initial, error_class, label_suffix,
+            empty_permitted, instance, use_required_attribute,
+            renderer,
+        )
+
+        for field in self.fields:
+            self.fields[field].required = False
+            # will cause BaseForm._clean_fields() to reuse the value
+            # from self.initial (<-- self.instance) if not specified
+            self.fields[field].disabled = field not in data
+
     def clean_status(self):
         return self.cleaned_data.get('status')
 
+    # this is copied from BaseRunForm
+    # TODO: come up with a single place for this, once NewForm has been refactored to ModelForm
+    def populate(self, product_id):
```

Comment:
Remove the TODOs, if anything the existing tools already mark these 2 lines and we definitely don't need more alerts from the rest of the tools.

FTR I don't think it's worth trying to refactor a 1 line method because the context is slightly different almost everywhere it is used.